userspace/libsinsp/stats.cpp

```diff
@@ -16,64 +16,406 @@ limitations under the License.
+		get_rss_vsz_pss_total_memory_and_open_fds(rss, vsz, pss, memory_used_host, open_fds_host);
+
+		buffer[SINSP_RESOURCE_UTILIZATION_CPU_PERC].value.d = cpu_usage_perc;
+		buffer[SINSP_RESOURCE_UTILIZATION_MEMORY_RSS].value.u32 = rss;
+		buffer[SINSP_RESOURCE_UTILIZATION_MEMORY_VSZ].value.u32 = vsz;
+		buffer[SINSP_RESOURCE_UTILIZATION_MEMORY_PSS].value.u32 = pss;
+		buffer[SINSP_RESOURCE_UTILIZATION_CONTAINER_MEMORY].value.u64 = get_container_memory_usage();
+		buffer[SINSP_RESOURCE_UTILIZATION_CPU_PERC_TOTAL_HOST].value.d = cpu_usage_perc_total_host;
+		buffer[SINSP_RESOURCE_UTILIZATION_PROCS_HOST].value.u32 = procs_running_host;
+		buffer[SINSP_RESOURCE_UTILIZATION_MEMORY_TOTAL_HOST].value.u64 = memory_used_host;
+		buffer[SINSP_RESOURCE_UTILIZATION_FDS_TOTAL_HOST].value.u64 = open_fds_host;
+
+		*nstats = SINSP_RESOURCE_UTILIZATION_FDS_TOTAL_HOST + 1;
+
+	}
+
+	if((flags & PPM_SCAP_STATS_STATE_COUNTERS) && stats_v2)
+	{
+		if(buffer[SINSP_STATS_V2_N_THREADS].name != nullptr && strncmp(buffer[SINSP_STATS_V2_N_THREADS].name, sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_THREADS], 10) != 0)
+		{
+			// Init
+			for(uint32_t i = SINSP_STATS_V2_N_THREADS; i < SINSP_MAX_STATS_V2; i++)
+			{
+				buffer[i].flags = PPM_SCAP_STATS_STATE_COUNTERS;
+				strlcpy(buffer[i].name, sinsp_stats_v2_resource_utilization_names[i], STATS_NAME_MAX);
+			}
+
+			buffer[SINSP_STATS_V2_NONCACHED_FD_LOOKUPS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_CACHED_FD_LOOKUPS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_FAILED_FD_LOOKUPS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_ADDED_FDS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_REMOVED_FDS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_STORED_EVTS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_STORE_EVTS_DROPS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_RETRIEVED_EVTS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_RETRIEVE_EVTS_DROPS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_NONCACHED_THREAD_LOOKUPS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_CACHED_THREAD_LOOKUPS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_FAILED_THREAD_LOOKUPS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_ADDED_THREADS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_REMOVED_THREADS].type = STATS_VALUE_TYPE_U64;
+			buffer[SINSP_STATS_V2_N_DROPS_FULL_THREADTABLE].type = STATS_VALUE_TYPE_U32;
+			buffer[SINSP_STATS_V2_N_MISSING_CONTAINER_IMAGES].type = STATS_VALUE_TYPE_U32;
+			buffer[SINSP_STATS_V2_N_CONTAINERS].type = STATS_VALUE_TYPE_U32;
+
+		}
+
+		// Get stats / metrics snapshot
+
+		buffer[SINSP_STATS_V2_N_THREADS].value.u64 = thread_manager->get_thread_count();
+		buffer[SINSP_STATS_V2_N_FDS].value.u64 = 0;
+		threadinfo_map_t* threadtable = thread_manager->get_threads();
+		threadtable->loop([&] (sinsp_threadinfo& tinfo) {
+			sinsp_fdtable* fdtable = tinfo.get_fd_table();
```

Comment:
here we need a check `sinsp_fdtable != nullptr`

Reply:
:heart: more segfaults is the last thing we need, which makes we wonder if we could come up with a more proper project inspection in this regard as sometimes we say we trust the pointer to be never NULL and prefer to save the overhead of checking vs sometimes we definitely need to check ...